### PR TITLE
Iter4 search ziyilin

### PR DESCRIPTION
**Pull request by CallipolisLin**

### CODE REVIEW

**Comment by Xiang-ruiqing** [comment]
Also need to deal with extra spaces within two words in a phrase, like "City           Dog"
Other possible reaction of searching a blank keyword rather than popping all results out

**Comment by Chenlibo** [comment]
> Also need to deal with extra spaces within two words in a phrase, like "City Dog"
> Other possible reaction of searching a blank keyword rather than popping all results out

I think spaces within two words are fine since "City Dog" usually means one thing instead of "City" and "Dog"

File: app/controllers/search_controller.rb [hunk 1]
```diff
@@ -5,33 +5,33 @@ def search
...
     filters = all_filters.select {|filter| params[filter]} 
     session[:filters] = filters
-
     if filters.empty?
       redirect_to results_path(:keyword => keyword), alert: "Please choose at least one category"
     elsif keyword.empty?
         redirect_to results_path(:keyword => ''), alert: "Please enter a keyword in the search box"
     else
-      flash[:notice] = "Search '#{keyword}' in #{filters.map{|x| x.inspect}.join(', ')}"
+      flash[:notice] = "Search '#{keyword.strip}' in #{filters.map{|x| x.inspect}.join(', ')}"
```
Comment (Chenlibo) [inline app/controllers/search_controller.rb]: nice

Comment (Xiang-ruiqing) [reply app/controllers/search_controller.rb]: I agree.

File: app/controllers/search_controller.rb [hunk 2]
```diff
@@ -5,33 +5,33 @@ def search
...
       redirect_to results_path(:keyword => keyword)
     end
   end
 
   def results
-    keyword = ("%" + params["keyword"] + "%").downcase
-    filters = session[:filters]
+    keyword = params["keyword"]
+    @filters = session[:filters] || all_filters
```
Comment (Xiang-ruiqing) [inline app/controllers/search_controller.rb]: Fine

**Pull request merged**